app/assets/javascripts/jquery.compose_export_dialog.js

```diff
@@ -0,0 +1,74 @@
+(function($){
+  $.PMX.ApplicationComposeExporter = function(el, options) {
+    var base = this,
+      composeDialog;
+
+    base.$el = $(el);
+
+    base.defaultOptions = {
+      composePath: 'data-composePath',
+      composeExportSelector: 'ul.application-button-menu a.export'
+    };
+
+    base.init = function() {
+      base.options = $.extend({}, base.defaultOptions, options);
+      base.bindEvents();
+    };
+
+    base.bindEvents = function() {
+      base.$el.on('click', base.options.composeExportSelector, base.triggerExport);
+    };
+
+    base.triggerExport = function(e) {
+      var $target = $(e.currentTarget);
+      var composeUrl = $target.attr(base.options.composePath);
+
+      e.preventDefault();
+      $.ajax({
+        type: 'GET',
+        headers: {
+          'Accept': 'application/json'
+        },
+        url: composeUrl
+      })
+        .done(function(response) {
+          composeDialog = base.initiateDialog(response.compose_yaml);
+          composeDialog.dialog('open');
+
+          $('.clipboard-copy').attr('data-clipboard-text', $('#composeYaml').text());
```

Comment:
The OCD in me does not like all the hard DOM dependencies, but I have a feeling you don't want to look at these lines of code anymore, :wink: